extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -1138,16 +1138,26 @@ export class Bind {
 
       updates.forEach(update => {
         const {boundProperty, newValue} = update;
+        const {property} = boundProperty;
         const mutation = this.applyBinding_(boundProperty, element, newValue);
+
         if (mutation) {
           mutations[mutation.name] = mutation.value;
-          const {property} = boundProperty;
           if (property == 'width') {
             width = isFiniteNumber(newValue) ? Number(newValue) : width;
           } else if (property == 'height') {
             height = isFiniteNumber(newValue) ? Number(newValue) : height;
           }
         }
+
+        if (isPropertyAFormValue(element.tagName, property)) {
+          const dispatchAt =
+            element.tagName === 'OPTION' ? element.closest('SELECT') : element;
+
+          if (dispatchAt) {
+            dispatchAmpFormValueChangeEvent(this.localWin_, dispatchAt);
+          }
+        }
```

Comment:
Nit: Can we keep everything in a single class function? It would also save you from checking `tagName` twice.

```js
this.dispatchFormValueChangeEventIfNecessary_(element, property);
```

Reply:
To only check `tagName` once, the `switch` gets a bit intense with the nested `if`s and early returns: https://gist.github.com/xrav3nz/6fd7b048080aedbaa1e87c418b4193e4 Are you okay with this?

Reply:
True, probably better as is. Speaking of switch statements, a lookup map might be more readable. 

Optional:
```js
/** @const {!Object<string, !Array<string>>} */
const FORM_VALUE_PROPERTIES = {
  'INPUT': ['checked', 'value'],
  'OPTION': ['selected'],
  'TEXTAREA': ['text'],
};

dispatchFormValueChangeEventIfNecessary_(element, property) {
  const props = FORM_VALUE_PROPERTIES[element.tagName];
  if (props && props.includes(property)) {
    ...
  }
}
```

Reply:
I think I'm gonna to leave it as is. Thanks again for the reviews! 💪